pkgs/bundle-util/default.nix

```diff
@@ -0,0 +1,98 @@
+{
+  /*
+    This utility creates a module that implements a bundle, which has
+    these features:
+    1. if the bundle is enabled, it enables by default a list of submodules
+    2. exposes its submodules as a list of module ID strings via the `submodules`
+    option
+
+    Example:
+
+    pkgs/bundles/nodejs/default.nix
+    ```nix
+    { pkgs, config, ... }:
+    pkgs.lib.mkBundleModule {
+    id = "nodejs";
+    name = "Node.js Tools Bundle";
+    description = "Development tools for the Node.js JavaScript runtime";
+    submodules = [
+    "interpreters.nodejs"
+    "languageServers.typescript-language-server"
+    "debuggers.node-dap"
+    "formatters.prettier"
+    "packagers.nodejs-packager"
+    ];
+    inherit pkgs config;
+    }
+    ```
+  */
+  mkBundleModule = { id, name, description, submodules, pkgs, config }:
+    with pkgs.lib;
+    let
+      cfg = config.bundles.${id};
+      /*
+      addEnableConfig - generates an enable config defaulted to true
+        for a submodules identified as 'elems' and merges it with an existing config.
+      elems - result of spliting a module ID on "." e.g. ["interpreters" "nodejs"]
+      config - an existing nested config, e.g.
+        {
+          interpreters = {
+            nodejs = {
+              enable = mkDefault true;
+            };
+          };
+        }
+      */
+      addEnableConfig = elems: config:
```

Comment:
```suggestion
      enablePath = elems: config:
```

Reply:
in general i think this function would be better implemented via https://noogle.dev/f/lib/attrsets/zip to ensure that the behavior is consistent with other modules' behavior

Reply:
i'd much rather see this function implemented in terms of https://noogle.dev/f/lib/setAttrByPath tbh

Reply:
After calling setAttrByPath you'd still need to traverse the graph a second pass to merge it with the other config. Doable but I'd have to write another function to do such merge. It's a tradeoff. I don't see a problem with the current implementation.